packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/pipeline-ai.spec.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import type { ComponentProps } from 'react';
+import {
+  cleanup,
+  fireEvent,
+  render,
+  screen,
+  waitFor,
+} from '@testing-library/react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import type { SinonSpy } from 'sinon';
+import { Provider } from 'react-redux';
+import preferencesAccess from 'compass-preferences-model';
+
+import { PipelineAI } from './pipeline-ai';
+import configureStore from '../../../../test/configure-store';
+import {
+  AIPipelineActionTypes,
+  changeAIPromptText,
+} from '../../../modules/pipeline-builder/pipeline-ai';
+
+const noop = () => {
+  /* no op */
+};
+
+const renderPipelineAI = ({
+  ...props
+}: Partial<ComponentProps<typeof PipelineAI>> = {}) => {
+  const store = configureStore();
+
+  render(
+    <Provider store={store}>
+      <PipelineAI onClose={noop} show {...props} />
+    </Provider>
+  );
+  return store;
+};
+
+const feedbackPopoverTextAreaId = 'feedback-popover-textarea';
+
+describe('PipelineAI Component', function () {
+  let store: ReturnType<typeof configureStore>;
+  afterEach(cleanup);
+
+  describe('when rendered', function () {
+    let onCloseSpy: SinonSpy;
+    beforeEach(function () {
+      onCloseSpy = sinon.spy();
+      store = renderPipelineAI({
+        onClose: onCloseSpy,
+      });
+    });
+
+    it('calls to close robot button is clicked', function () {
+      expect(onCloseSpy.called).to.be.false;
+      const closeButton = screen.getByTestId('close-ai-button');
+      expect(closeButton).to.be.visible;
+      closeButton.click();
+      expect(onCloseSpy.calledOnce).to.be.true;
+    });
+  });
+
+  describe('when rendered with text', function () {
+    beforeEach(function () {
+      store = renderPipelineAI();
+      store.dispatch(changeAIPromptText('test'));
+    });
+
+    it('calls to clear the text when the X is clicked', function () {
+      expect(store.getState().pipelineBuilder.aiPipeline.aiPromptText).to.equal(
+        'test'
+      );
+
+      const clearTextButton = screen.getByTestId('ai-text-clear-prompt');
+      expect(clearTextButton).to.be.visible;
+      clearTextButton.click();
+
+      expect(store.getState().pipelineBuilder.aiPipeline.aiPromptText).to.equal(
+        ''
+      );
+    });
+  });
+
+  describe('Query AI Feedback', function () {
```

Comment:
Pipeline AI Feedback?

Reply:
In the compass tracking plan, it is called "AI Response Rated" and it is one kind of event for both query and aggregation that has the `editor_view_type: find | stage | text ` prop to distinguish what we leave feedback for. The plan also uses different values for feedback. It is actually called `rating` with `good | bad | dismissed` values. Should we update this format in both places, here and in the query bar?

```
AI Response Rated

editor_view_type: find | stage | text 
query_shape: <stages & operators used>
rating: good | bad | dismissed
num_tokens: <length of repsonse in number of tokens>
```

Reply:
Yes, let's do that in a separate pr.